### PR TITLE
chore: add duplex iterable type to connection

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR adds the duplex iterable type to connection per https://github.com/libp2p/js-libp2p/pull/802#discussion_r524290682

This already existed in the pubsub peer streams and it was moved into the conenction and linked from there

cc @wemeetagain (could not ask you direct review)

### CODE REVIEW

File: src/connection/connection.d.ts [hunk 1]
```diff
@@ -1,5 +1,24 @@
 declare const _exports: typeof Connection;
 export = _exports;
+export type Sink = (source: Uint8Array) => Promise<Uint8Array>;
+export type DuplexIterableStream = {
+    sink: Sink;
+    source: () => AsyncIterator<Uint8Array, any, undefined>;
+};
```
Comment (jacobheun) [inline src/connection/connection.d.ts:7]: The type here should really just be a `Stream`, and it should match what's coming from mplex https://github.com/libp2p/js-libp2p-mplex/blob/v0.10.1/src/stream.js#L59

Comment (vasco-santos) [reply src/connection/connection.d.ts:7]: Should we add the Stream in a type declaration file in the multiplexer interface?

File: src/connection/connection.js [hunk 2]
```diff
@@ -1,56 +1,23 @@
...
-  }
-}
+/**
+ * @callback Sink
+ * @param {Uint8Array} source
+ * @returns {Promise<Uint8Array>}
+ *
+ * @typedef {object} DuplexIterableStream
+ * @property {Sink} sink
+ * @property {() AsyncIterator<Uint8Array>} source
```
Comment (Gozala) [inline src/connection/connection.js:16]: Is this `AsyncIterator` or is it meant to be `AsyncIterable` ?

**Comment by vasco-santos** [comment]
Closed in favour of #74

**Pull request closed without merging**